### PR TITLE
Feature/fix ukrlp name

### PR DESCRIPTION
**Pull request by nshumoogum**
### What

Update business logic to retrieve the institution name from the `ProviderAliases` field in xml unless the field does not exist or is empty, at which point fall back to the `ProviderName`.

### How to review

1) Check code logic
2) Check tests pass

### CODE REVIEW

File: CreateUkrlpBlobTrigger/helper.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    @staticmethod
+    def get_provider_name(matching_provider_records):
+        """Returns a single name for the provider"""
+
+        # Get provider name if alias does not exist in ukrlp response body
+        if 'ProviderAliases' in matching_provider_records and matching_provider_records['ProviderAliases'] is not None:
+            aliases = Helper.get_list(matching_provider_records['ProviderAliases'])
+            return aliases[0]['ProviderAlias']
+        else:
```
Comment (philratcliffe) [inline CreateUkrlpBlobTrigger/helper.py]: Maybe remove else as unnecessary and then consistent with get_list above too.

File: CreateUkrlpBlobTrigger/helper.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            return matching_provider_records['ProviderName']
+
+
```
Comment (philratcliffe) [inline CreateUkrlpBlobTrigger/helper.py]: Maybe remove extra linefeed.

File: CreateUkrlpBlobTrigger/lookup_creator.py [hunk 3]
```diff
@@ -66,7 +67,10 @@ def get_lookup_entry(ukprn, matching_provider_records):
         lookup_item['id'] = utils.get_uuid()
         lookup_item['created_at'] = datetime.datetime.utcnow().isoformat()
         lookup_item['ukprn'] = ukprn
-        lookup_item['ukprn_name'] = matching_provider_records['ProviderName']
+        logging.info(f'give example of provider record {str(matching_provider_records)}')
+
```
Comment (philratcliffe) [inline CreateUkrlpBlobTrigger/lookup_creator.py]: Do you need the str?

File: CreateUkrlpBlobTrigger/tests/test_helper.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        expected_key = 'Canton University'
+        multipleAliases = {'ProviderName': 'University of England', 'ProviderAliases': [{'ProviderAlias': 'Canton University'}, {'ProviderAlias: Roath College'}, {'ProviderAlias': 'Whitchurch Engineering College'}]}
+        
+        key = Helper.get_provider_name(multipleAliases)
+        self.assertEqual(expected_key, key)
+
+    def test_when_there_is_only_one_alias(self):
+        expected_key = 'Roath College'
+        singleAlias = {'ProviderName': 'University of England',            'ProviderAliases': [{'ProviderAlias': 'Roath College'}]}
+
```
Comment (philratcliffe) [inline CreateUkrlpBlobTrigger/tests/test_helper.py]: Looks like additional spaces after comma.

File: CreateUkrlpBlobTrigger/tests/test_helper.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        emptyProviderAliases = {'ProviderName': 'University of England', 'ProviderAliases': None}
+
+        key = Helper.get_provider_name(emptyProviderAliases)
+        self.assertEqual(expected_key, key)
+
+    def test_when_provider_aliases_is_missing(self):
+        expected_key = 'University of Wales'
+        missingProviderAliases = {'ProviderName': 'University of Wales'}
+
+        key = Helper.get_provider_name(missingProviderAliases)
```
Comment (philratcliffe) [inline CreateUkrlpBlobTrigger/tests/test_helper.py]: Missing newline?

File: CreateUkrlpBlobTrigger/lookup_creator.py (hunk 3 above)
Comment (nshumoogum) [reply CreateUkrlpBlobTrigger/lookup_creator.py]: Sorry this log line should be removed

File: CreateUkrlpBlobTrigger/tests/test_helper.py (hunk 5 above)
Comment (nshumoogum) [reply CreateUkrlpBlobTrigger/tests/test_helper.py]: Will do

**Pull request merged**